test/test_model.py

```diff
@@ -1174,6 +1174,100 @@ def test_recursive_push_and_pull(case):
         assert c.exists(fpath), f'pull_path failed: file {fpath} missing at destination'
 
 
+def change_to_relative_path(case: PushPullCase):
```

Comment:
This is a ton of code just to test this case. Instead of trying to re-run all the test cases above, can we just test this specific case? I've been playing with it locally, and this seems to exercise it:

```python
def test_push_path_relative():
    harness = ops.testing.Harness(ops.CharmBase, meta='''
        name: test-app
        containers:
          foo:
            resource: foo-image
        ''')
    harness.begin()
    harness.set_can_connect('foo', True)
    container = harness.model.unit.containers['foo']

    with tempfile.TemporaryDirectory() as tmpdir:
        tmp = pathlib.Path(tmpdir)
        os.makedirs(tmp / 'src' / 'foo')
        with open(tmp / 'src' / 'foo' / 'bar.txt', 'w') as f:
            f.write('hello')

        rel_src = os.path.relpath(tmp / 'src' / 'foo' / 'bar.txt')
        container.push_path(rel_src, tmp / 'dest')

        content = container.pull(tmp / 'dest' / 'bar.txt').read()
        assert content == 'hello'
```

Reply:
I'm not quite sure that using a relative pathing would really test the case that I'm trying to fix.
The relative path would give a path like: `../../path/to/dir`, whereas the test case that i'm trying to test is `path/to/dir`, for example in charm src directory, when trying to push contents under `templates/`.

This goes to explain the addition of `prefix != '.'` code in the previous comment.
When using a path under the charm src, using the example `templates/` path, the prefix would be set as `.`, as it tries to get the parent of the `templates/` path. 

Here's a code snippet to replicate the behavior: https://pastebin.canonical.com/p/yHdMN4xSZz/

Reply:
Okay. The test I included definitely failed before the fix and works after, so it's doing something. :-)

But if we need to use a path without a dot prefix as well, like `templates/config.yaml`, that's fine. Let's just keep the test simple and focussed. I don't mind using `os.chdir` if we need to -- though if we do that, let's do it in a `try ... finally` block to change back to the old directory even in case of failure.

Reply:
Sounds good! Thank you for the guidance!

Reply:
Refactored the tests, thank you!